File: server/matchmaking/src/main/java/com/boxtrotstudio/ghost/matchmaking/network/packets/UpdateSessionPacket.java
package com.boxtrotstudio.ghost.matchmaking.network.packets;

import com.boxtrotstudio.ghost.matchmaking.network.PlayerClient;
import com.boxtrotstudio.ghost.matchmaking.network.TcpServer;
import com.boxtrotstudio.ghost.network.packet.Packet;

import java.io.IOException;

public class UpdateSessionPacket extends Packet<TcpServer, PlayerClient> {
    public UpdateSessionPacket(PlayerClient client) {
        super(client);
    }

    @Override
    public void onWritePacket(PlayerClient client, Object... args) throws IOException {

        write((byte) 0x29)
                .write(client.getPlayer().getSession().length())
                .write(client.getPlayer().getSession())
                .endTCP();
    }
}

File: client/core/src/com/boxtrotstudio/ghost/client/network/packets/UpdateSessionPacket.kt
package com.boxtrotstudio.ghost.client.network.packets

import com.boxtrotstudio.ghost.client.Ghost
import com.boxtrotstudio.ghost.client.network.Packet
import com.boxtrotstudio.ghost.client.network.PlayerClient


class UpdateSessionPacket : Packet<PlayerClient>(){
    override fun handle(){
        val newSession = consume(consume(4).asInt()).asString()

        Ghost.Session = newSession
    }
}
